pkgs/test_core/lib/src/runner/engine.dart

```diff
@@ -293,6 +302,29 @@ class Engine {
     return success;
   }
 
+  Future<Null> _gatherCoverage(LiveSuiteController controller) async {
+    if (_coverage == null) return;
+
+    final RunnerSuite suite = controller.liveSuite.suite;
+
+    if (!suite.platform.runtime.isDartVM) return;
+
+    // for some reason the observatoryUrl URI doesn't have the query parameters populated
+    final String isolateId = Uri.splitQueryString(
+        suite.environment.observatoryUrl.toString().split('?')[1])['isolateId'];
+
+    final Map<String, dynamic> cov = await collect(
+        suite.environment.observatoryUrl, false, false, false, Set(),
```

Comment:
[nit] Dart `2.2.0` should support Set literals.

```suggestion
        suite.environment.observatoryUrl, false, false, false, {},
```

Reply:
Done!

Reply:
This one failed analysis, I guess the empty Set literally wasn't quite perfect at `2.2.0`. Put it back.